components/Account/AuthSettings.js

```diff
@@ -0,0 +1,149 @@
+
+    return (
+      <Loader
+        loading={loading}
+        error={error}
+        render={() => {
+          const { enabledFirstFactors, preferredFirstFactor } = me
+          const selectedFirstFactor = preferredFirstFactor || DEFAULT_TOKEN_TYPE
+          const { mutating, serverError } = this.state
+
+          return (
+            <Fragment>
+              <fieldset {...styles.fieldset}>
+                <legend>
+                  {t('account/authSettings/firstfactor/label')}{' '}
+                  {mutating && (
+                    <span {...styles.spinnerWrapper}>
+                      <InlineSpinner size={24} />
+                    </span>
+                  )}
+                </legend>
+                {SUPPORTED_TOKEN_TYPES.map((tokenType) => {
+                  const disabled = enabledFirstFactors.indexOf(tokenType) === -1
+                  return (
+                    <p key={tokenType}>
+                      <Radio
+                        checked={tokenType === selectedFirstFactor}
+                        disabled={disabled || mutating}
+                        onChange={(_, checked) => {
+                          this.setState(state => ({
+                            mutating: true
+                          }))
+                          const finish = () => {
+                            this.setState(state => ({
+                              mutating: false
+                            }))
+                          }
+                          updatePreferredFirstFactor({
+                            tokenType
+                          })
+                            .then(finish)
+                            .catch(this.catchServerError)
+                        }}
+                      >
+                        {t(`account/authSettings/firstfactor/${tokenType}/label${disabled ? '/disabled' : ''}`)}
+                      </Radio>
+                    </p>
+                  )
+                })}
+              </fieldset>
+              {serverError && <ErrorMessage error={serverError} />}
+            </Fragment>
+          )
+        }}
+      />
+    )
+  }
+}
+
+const mutation = gql`
+  mutation preferredFirstFactor(
+    $tokenType: SignInTokenType
+  ) {
+    preferredFirstFactor(tokenType: $tokenType) {
+      id
+      enabledFirstFactors
+      preferredFirstFactor
+    }
+  }
+`
+
+const query = gql`
+  query myAuthSettings {
+    me {
+      id
+      enabledFirstFactors
+      preferredFirstFactor
+    }
+  }
+`
+
+export default compose(
+  graphql(mutation, {
+    props: ({ mutate }) => ({
+      updatePreferredFirstFactor: ({ tokenType }) =>
+        mutate({
+          variables: {
+            tokenType
+          }
+        })
+    })
+  }),
+  graphql(query, {
+    props: ({ data, errors }) => ({
+      data,
+      loading: data.loading || !data.me,
+      error: data.error,
+      me: data.loading ? undefined : data.me
```

Comment:
it is misleading to call this `me`. Call it `authSettings` or directly map the two variables (`enabledFirstFactors` and `preferredFirstFactor`)

Reply:
@uxengine: fyi 
@tpreusse: as he is in holidays, I would appreciate if you can clean it up.